src/serial/xterm-keyboard.ts

```diff
@@ -12,6 +12,24 @@ const copyShortcut = (e: KeyboardEvent): boolean => {
   return true;
 };
 
+let escWasPressed = false;
+let escTimeout: NodeJS.Timeout;
+const tabOutShortcut = (e: KeyboardEvent, tabTo: HTMLElement): boolean => {
+  if (e.code === "Escape") {
+    clearTimeout(escTimeout);
```

Comment:
Shouldn't we also clear the timeout for non-escape key presses (including tab)?

Reply:
I've tweaked this to clear the timeout and reset `escWasPressed` to `false` if any key other than `escape` was pressed. Pressing `escape` should just clear and reset the timeout. This means pressing `escape`, `*any other key*` , then `tab` will not exit the terminal. [`9d5c827`](https://github.com/microbit-foundation/python-editor-next/pull/942/commits/9d5c82796d502173b7ceb681a8db7e5b70e1f2d6)

Reply:
Thanks, this looks good.

Reply:
I want to have a quick play and convince myself the React context stuff is necessary, otherwise this looks good.